examples/jsm/nodes/accessors/ReflectNode.js

```diff
@@ -44,29 +46,25 @@ ReflectNode.prototype.generate = function ( builder, output ) {
 
 			case ReflectNode.VECTOR:
 
-				builder.addNodeCode( 'vec3 reflectVec = inverseTransformDirection( reflect( -normalize( vViewPosition ), normal ), viewMatrix );' );
-
-				result = 'reflectVec';
+				var normal = this.normal.build( builder, 'v3' );
+				
+				result = 'inverseTransformDirection( reflect( -normalize( vViewPosition ), ' + normal + ' ), viewMatrix )';
 
 				break;
 
 			case ReflectNode.CUBE:
 
-				var reflectVec = new ReflectNode( ReflectNode.VECTOR ).build( builder, 'v3' );
-
-				builder.addNodeCode( 'vec3 reflectCubeVec = vec3( -1.0 * ' + reflectVec + '.x, ' + reflectVec + '.yz );' );
+				var reflectVec = new ReflectNode( ReflectNode.VECTOR, this.normal ).build( builder, 'v3' );
 
-				result = 'reflectCubeVec';
+				result = 'vec3( -1.0 * ' + reflectVec + '.x, ' + reflectVec + '.yz )';
 
 				break;
 
 			case ReflectNode.SPHERE:
 
-				var reflectVec = new ReflectNode( ReflectNode.VECTOR ).build( builder, 'v3' );
-
-				builder.addNodeCode( 'vec2 reflectSphereVec = normalize( ( viewMatrix * vec4( ' + reflectVec + ', 0.0 ) ).xyz + vec3( 0.0, 0.0, 1.0 ) ).xy * 0.5 + 0.5;' );
```

Comment:
This is used in together with `unique` to share the same code of the class and not only of node. For example:

```
material.color = new ReflectNode();
material.emissive = new ReflectNode();
// if is share he will create two codes for each slot
// if is unique he will share the same code for each slot
```

Reply:
The reason I changed it is because it is no longer unique. Some times we reflect normal, some times we reflect clearCoatNormals. I tough making a CleatCoatReflectNode was too much code duplication.